modules/v1/workspaces/controllers/ArticleFileController.php

```diff
@@ -0,0 +1,178 @@
+        $articleId = $request->get('articleId');
+
+        $query = ArticleFileResource::find()->byArticleId($articleId);
+
+        return new ActiveDataProvider([
+            'query' => $query,
+            'pagination' => false
+        ]);
+    }
+
+    /**
+     * Upload files
+     *
+     * @return mixed
+     * @throws Exception
+     * @throws ValidationException
+     * @throws \yii\base\Exception
+     */
+    public function actionAttachFiles()
+    {
+        $request = Yii::$app->request;
+        $article = ArticleResource::findOne($request->post('article_id'));
+
+        if (!$article) {
+            throw new ValidationException(Yii::t('app', 'Article not found'));
+        }
+
+        $files = UploadedFile::getInstancesByName('files');
+        if (!$files) {
+            throw new ValidationException(Yii::t('app', 'Unable to find files'));
+        }
+
+        $dbTransaction = Yii::$app->db->beginTransaction();
+
+        $attachments = [];
+
+        foreach ($files as $file) {
+            $articleFile = ArticleFileResource::find()
+                ->byArticleId($article->id)
+                ->byName($file->name)
+                ->one();
+
+            // if exist same name file overwrite
+            if (!$articleFile) {
+                $articleFile = new ArticleFileResource();
+                $articleFile->article_id = $article->id;
+            }
+
+            if (!$articleFile->uploadFile($file)) {
+                $dbTransaction->rollBack();
+                throw new ValidationException(Yii::t('app', 'Unable to upload attachment'));
+            }
+
+            // TODO non mp4 videos to be converted into mp4
+
+            if (!$articleFile->save()) {
+                $dbTransaction->rollBack();
+                throw new ValidationException(Yii::t('app', 'Unable to save attachment'));
+            }
+
+            $attachments[] = $articleFile;
+        }
+        $dbTransaction->commit();
+
+        return $this->response($attachments, 201);
+    }
+
+    /**
+     * Get attachment config data
+     *
+     * @return mixed
+     */
+    public function actionGetAttachConfig()
+    {
+        $phpConfig = [
+            'upload_max_filesize' => ['title' => 'Max file size', 'size' => ini_get('upload_max_filesize')],
+            'max_file_uploads' => ['title' => 'Max file uploads', 'size' => ini_get('max_file_uploads')],
+        ];
+
+        return $this->response($phpConfig);
+    }
+
+    /**
+     * Change article file label
+     *
+     * @return ArticleFileResource|array
+     * @throws ValidationException
+     */
+    public function actionChangeLabel()
+    {
+        $request = Yii::$app->request;
+
+        $articleFile = ArticleFileResource::findOne(['id' => $request->post('id')]);
+
+        if (!$articleFile) {
+            throw new ValidationException(Yii::t('app', 'Article file not exist'));
+        }
+
+        if (!$articleFile->load($request->post(), '') && !$articleFile->save()) {
```

Comment:
@skavt This is not correct, because it first statement returns false, then it won't enter the if statement and action will return 200. U should use `||` operator